lib/chip-types/field.js

```diff
@@ -17,7 +17,7 @@ function Field (declaration){
 		this[method_name] = (function(method_name){
 			return function(){
 				const arguments_array = Object.keys(arguments).map((key)=>arguments[key]);
-				arguments_array.splice(1, 0, this.params);
+				arguments_array.splice(1, 0, self.params);
```

Comment:
I agree, it's kind of mysterious. It works cool, but we have to refactor it sometime